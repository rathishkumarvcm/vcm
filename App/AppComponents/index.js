import CustomPageWizard from './CustomPageWizard';
import CustomRadio from './CustomRadio';
import CustomCheckBox from './CustomCheckBox';
import CustomDropDown from './CustomDropDown';



export {
    CustomPageWizard,
    CustomRadio,
    CustomCheckBox,
    CustomDropDown

};