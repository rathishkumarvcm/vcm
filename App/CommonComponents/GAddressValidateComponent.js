import React from 'react';
import { View, StyleSheet, Text } from 'react-native';
import globalString from '../Constants/GlobalStrings';
import GInputComponent from './GInputComponent';
import PropTypes from "prop-types";
import { scaledHeight } from '../Utils/Resolution';

export const styles = StyleSheet.create({
    settingsView1: {
        marginTop: scaledHeight(18),
        paddingLeft: '4%',
        paddingRight: '4%',
    },
    editAddressTypeLabel: {
        color: '#333333DE',
        fontSize: scaledHeight(16),
        fontWeight: 'bold',
        marginBottom: '4%'
    },
    editAddressInput: {
        flexDirection: 'column',
        width: '100%',
        paddingLeft: '2%',
        paddingRight: '2%',
        marginBottom: '2%',
        justifyContent: 'center'
    },
    editFlexDirectionColumn: {
        flexDirection: 'column',
        width: '100%',
        marginTop: scaledHeight(20),
        paddingLeft: '4%',
        paddingRight: '4%'
    },
    editAddressCityLabel: {
        color: '#333333DE',
        fontWeight: 'bold',
        margin: '2%',
        fontSize: scaledHeight(18)
    },
    editAddressCityValue: {
        color: '#56565A',
        margin: '2%',
        fontSize: scaledHeight(18)
    },
    saveButtonStyle: {
        alignItems: 'center',
        backgroundColor: '#56565A',
        borderColor: '#56565A',
        borderWidth: 1,
        height: scaledHeight(50),
        justifyContent: 'center',
        marginLeft: '4%',
        marginRight: '4%',
        marginTop: scaledHeight(12),
        width: '92%'
    },
    saveButtonText: {
        color: '#FFFFFF',
        fontSize: scaledHeight(16)
    },
});

export const GAddressValidateComponent = (props) => (
    <View>
        <View style={styles.settingsView1}>
            <Text style={styles.editAddressTypeLabel}>
                {globalString.addAddressInfo.addressLineOne}
            </Text>
        </View>

        <View style={styles.editAddressInput}>
            <GInputComponent
                propInputStyle={styles.editAddressInput}
                placeholder={globalString.addAddressInfo.addressLineOne}
                onChangeText={props.setAddressOne}
                value={props.addressOne}
                maxLength={50}
                errorFlag={!props.validationAddressOne}
                errorText={globalString.profileValidationMessages.validateAddressLineOne}
            />
        </View>

        <View style={styles.settingsView1}>
            <Text style={styles.editAddressTypeLabel}>
                {globalString.addAddressInfo.addressLineTwo}
            </Text>
        </View>

        <View style={styles.editAddressInput}>
            <GInputComponent
                propInputStyle={styles.editAddressInput}
                placeholder={globalString.addAddressInfo.addressLineTwo}
                value={props.addressTwo}
                maxLength={50}
                onChangeText={props.setAddressTwo}
                errorFlag={!props.validationAddressTwo}
                errorText={globalString.profileValidationMessages.validateAddressLineTwo}
            />
        </View>

        <View style={styles.settingsView1}>
            <Text style={styles.editAddressTypeLabel}>
                {globalString.addAddressInfo.zipCode}
            </Text>
        </View>

        <View style={styles.editAddressInput}>
            <GInputComponent
                propInputStyle={styles.editAddressInput}
                placeholder={globalString.addAddressInfo.zipCode}
                onBlur={props.validateZipCodeValue}
                onChangeText={props.setZipcodeValue}
                value={props.zipCodeValue}
                keyboardType="number-pad"
                maxLength={5}
                errorFlag={!props.isZipCodeValid}
                errorText={globalString.profileValidationMessages.validateZipcode}
            />
        </View>

        <View style={styles.editFlexDirectionColumn}>
            <Text style={styles.editAddressCityLabel}>
                {globalString.addAddressInfo.cityLabel}
            </Text>
            <Text style={styles.editAddressCityValue}>
                {props.userCity}
            </Text>
        </View>

        <View style={styles.editFlexDirectionColumn}>
            <Text style={styles.editAddressCityLabel}>
                {globalString.addAddressInfo.stateLabel}
            </Text>
            <Text style={styles.editAddressCityValue}>
                {props.userState}
            </Text>
        </View>

        {!props.isValidateAddress && (
            <GButtonComponent
                buttonStyle={styles.saveButtonStyle}
                buttonText={gblStrings.common.validate}
                textStyle={styles.saveButtonText}
                onPress={props.validateAddressValue} />
        )}
    </View>
);

GAddressValidateComponent.propType = {
    addressOne: PropTypes.string,
    addressTwo: PropTypes.string,
    zipCodeValue: PropTypes.string,
    userCity: PropTypes.string,
    userState: PropTypes.string,

    validationAddressOne: PropTypes.bool,
    validationAddressTwo: PropTypes.bool,
    isZipCodeValid: PropTypes.bool,
    isValidateAddress: PropTypes.bool,

    setAddressOne: PropTypes.func,
    setAddressTwo: PropTypes.func,
    validateZipCodeValue: PropTypes.func,
    setZipcodeValue: PropTypes.func,
    validateAddressValue: PropTypes.func
};

GAddressValidateComponent.defaultProps = {
    addressOne: '',
    addressTwo: '',
    zipCodeValue: '',
    userCity: ' - ',
    userState: ' - ',

    validationAddressOne: true,
    validationAddressTwo: true,
    isZipCodeValid: true,
    isValidateAddress: true,

    setAddressOne: null,
    setAddressTwo: null,
    validateZipCodeValue: null,
    setZipcodeValue: null,
    validateAddressValue: null
};

export default GAddressValidateComponent;