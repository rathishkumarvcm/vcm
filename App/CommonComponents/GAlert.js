/* **************************\
Function: showAlertWithCancelButton.
Explanation:
Common method to show the popup with two buttons , title and message.
Each button will have its standalone call back
============================


\************************** */

import { Alert } from "react-native";

function showAlertWithCancelButton(
  title,
  text,
  okButtonLabel,
  cancelButtonLabel,
  okCallback,
  cancelCallback
) {
  return Alert.alert(
    title,
    text,
    [
      { text: okButtonLabel, onPress: okCallback },
      { text: cancelButtonLabel, onPress: cancelCallback }
    ],
    { cancelable: false }
  );
}

function showAlert(title, text, okButtonLabel, okCallback) {
  return Alert.alert(
    title,
    text,
    [{ text: okButtonLabel, onPress: okCallback }],
    { cancelable: false }
  );
}

export { showAlert, showAlertWithCancelButton };
