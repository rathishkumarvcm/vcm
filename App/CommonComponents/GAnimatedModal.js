import React from 'react';
import {Animated, Easing, Dimensions} from 'react-native';
import PropTypes from "prop-types";

const { height } = Dimensions.get('window');

const styles = {
    container: {
      position: "absolute",
      height: '50%',
      width: '100%',
      bottom: -height, //  initial position of the view. Hide the view at bottom.
      backgroundColor: "#fff",
    }
  };

export default class GAnimatedModal extends React.Component {
    constructor(props){
        super(props);
       const {containerStyle} = this.props;
        this.state={
        };

        this.yTranslate = new Animated.Value(0);
        this.negativeHeight = -height;
        this.modalMoveY = this.yTranslate.interpolate({
          inputRange: [0, 1],
          outputRange: [0, this.negativeHeight]
      });
      this.translateStyle = { transform: [{ translateY: this.modalMoveY }] }; 
      this.amimatedViewStyle = [styles.container, this.translateStyle,containerStyle];
    }

    static getDerivedStateFromProps(/* props, state */){
      // initialize state variable and return. If no changes required for state
      // variable then return empty object. return {}
      // console.log("props",props,state);
      return {}; // should return empty object by default
    }

      shouldComponentUpdate(nextProps){
        const {visible}=this.props;
        if(visible===nextProps.visible){
            return false;
        }
        return true;
      }

      componentDidUpdate(/* prevProps, prevState */) {
       // console.log("prevProps",prevProps,prevState);
        const {visible}=this.props;
          if (visible) {
            //  animate to show the modal
            this.yTranslate.setValue(0); //  reset the animated value
            Animated.spring(this.yTranslate, {
              toValue: 1,
              duration: 400,
              overshootClamping:true 
            }).start();
          } else {
            //  animate to hide the modal
            Animated.timing(this.yTranslate, {
              toValue: 0,
              duration: 400,
              easing: Easing.linear
            }).start();
          }
        }

    render(){
      const {children}=this.props;
        return (
            <Animated.View style={this.amimatedViewStyle}>
              {children}
            </Animated.View>
          );
    }
}

GAnimatedModal.propTypes = {
  visible : PropTypes.bool,
  children: PropTypes.instanceOf(Object),
  containerStyle: PropTypes.instanceOf(Object)
};

GAnimatedModal.defaultProps = {
  visible : false,
  children :{},
  containerStyle :{}

 
};