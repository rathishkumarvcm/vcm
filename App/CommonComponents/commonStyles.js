import { StyleSheet } from "react-native";
import { scaledHeight } from '../Utils/Resolution';


const createStyles = (props) => { 
    return StyleSheet.create({
        buttonStyle:{
          alignItems:'center',
          backgroundColor:"#56565A",
          borderRadius:1,
          height:scaledHeight(props.height),
          justifyContent: "center",
          marginTop:'2%',
          width:'90%',
          ...props.buttonStyle
    }
    });
    };

export default createStyles;