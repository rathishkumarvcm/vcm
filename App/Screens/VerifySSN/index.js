import VerifySSNComponent from './VerifySSNContainer';

export default VerifySSNComponent;