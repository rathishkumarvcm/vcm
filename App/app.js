import React, { Component } from 'react';
import { setJSExceptionHandler, setNativeExceptionHandler } from 'react-native-exception-handler';
import RootComponent from './RootComponent';
import AppUtils from './Utils/AppUtils';

const jsExceptionhandler = (error) => {
  //  This is your custom global error handler
  //  You do stuff like show an error dialog
  //  or hit google analytics to track crashes
  //  or hit a custom api to inform the dev team.

  AppUtils.debugLog('JS Exception handled:-', error);
};

//  //  - exceptionhandler is the exception handler function
//  //  - allowInDevMode is an optional parameter is a boolean.
//  //    If set to true the handler to be called in place of RED screen
//  //    in development mode also.
setJSExceptionHandler(jsExceptionhandler, true);



const nativeExceptionhandler = exceptionString => {
  //  your exception handler code here
  AppUtils.debugLog('Native Exception handled:-', exceptionString);
};
setNativeExceptionHandler(
  nativeExceptionhandler,
  false,
  false
);
//  - exceptionhandler is the exception handler function
//  - forceAppQuit is an optional ANDROID specific parameter that defines
//     if the app should be force quit on error.  default value is true.
//     To see usecase check the common issues section.
//  - executeDefaultHandler is an optional boolean (both IOS, ANDROID)
//     It executes previous exception handlers if set by some other module.
//     It will come handy when you use any other crash analytics module along with this one
//     Default value is set to false. Set to true if you are using other analytics modules.

export default class AppComponent extends Component {
  constructor(props) {
    super(props);
    //  set true to isLoading if data for this screen yet to be received and wanted to show loader.
    this.state = {
    };
  }

  render() {
    return (
      <RootComponent />
    );
  }

}