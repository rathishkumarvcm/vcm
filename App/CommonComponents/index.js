import GLoadingSpinner from './GLoadingSpinner';
import GButtonComponent from './GButtonComponent';
import GAnimatedModal from './GAnimatedModal';
import GInputComponent from './GInputComponent';
import GBiometricAuthentication from './GBiometricAuthentication';
import GFloatingInputComponent from './GFloatingInputComponent';
import GRadioButtonComponent from './GRadioButtonComponent';
import GCheckBoxComponent from './GCheckBoxComponent';
import GCardTileComponent from './GCardTileComponent';
import GFlatListView from './GFlatListView';
import GCounterComponent from './GCounterComponent';
import GCommonFlatList from './GCommonFlatList';
import GPagination from './GPagination';
import GWebViewComponent from './GWebViewComponent';
import GSearchComponent from './GSearchComponent';
import GIconButton from './GIconButton';
import GIcon from './GIcon';
import {showAlert,showAlertWithCancelButton} from './GAlert';
import GHeaderComponent from './GHeaderComponent';
import GFooterComponent from './GFooterComponent';
import GRatingStarsComponent from './GRatingStarsComponent';
import GSwitchComponent from './GSwitchComponent';
import GFooterSettingsComponent from './GFooterSettingsComponent';
import GDropDownComponent from './GDropDownComponent';
import GDateComponent from './GDateComponent';
import GModalComponent from './GModalComponent';
import GCollapseComponent from './GCollapseComponent';
import GSingletonClass from './GSingletonClass';
import GBankAccValidateComponent from './GBankAccValidateComponent';
import GAddressValidateComponent from './GAddressValidateComponent';

import {
    SetKeyValueToStore,
    GetKeyValueFromStore,
    RemoveKeyValueFromStore
  } from "./GPersistantStore";

import GErrorBoundaries from './GErrorBoundaries';

export {
    GLoadingSpinner,
    GButtonComponent,
    GAnimatedModal,
    GInputComponent,
    GBiometricAuthentication,
    GFloatingInputComponent,
    GRadioButtonComponent,
    GCheckBoxComponent,
    GCardTileComponent,
    GFlatListView,
    GCounterComponent,
    GCommonFlatList,
    GPagination,
    GWebViewComponent,
    GSearchComponent,
    GIconButton,
    GIcon,
    showAlert,
    showAlertWithCancelButton,
    GHeaderComponent,
    GFooterComponent,
    GRatingStarsComponent,
    SetKeyValueToStore,
    GetKeyValueFromStore,
    RemoveKeyValueFromStore,
    GSwitchComponent,
    GFooterSettingsComponent,
    GDropDownComponent,
    GDateComponent,
    GModalComponent,
    GCollapseComponent,
    GSingletonClass,
    GBankAccValidateComponent,
    GAddressValidateComponent,
    GErrorBoundaries
};