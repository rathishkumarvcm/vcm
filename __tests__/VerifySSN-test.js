import React from 'react';
import renderer from 'react-test-renderer';
import {shallow} from 'enzyme';
import VerifySSN from '../App/Screens/VerifySSN/VerifySSNComponent';
import { GButtonComponent } from '../App/CommonComponents';

describe('Verify SSN ',()=>{

  //  Snap shot 
  it('snap shot', () => {  
    const tree = renderer.create(<VerifySSN />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  //  Render using shallow - shallow not includes child component
  test('render',()=>{
    const component = shallow(<VerifySSN />);  
    component.render();
  });

  //  Function testing
  test('function testing',()=>{
    const component = shallow(<VerifySSN />);
    const wrapper = new VerifySSN();
    component.render();

    wrapper.onClickNext();
 });

 //  Input Component matches the length 
it('Input text matches with length', () => {
  const instanceOf = renderer.create(<VerifySSN />).getInstance();
  instanceOf.setSocialSecurityNumber('123456789');  
  expect(instanceOf.state.socialSecurityNumber).toHaveLength(9);
});

 //  Input Component matches with given input
it('Input text matches with given input',()=>{
  const instanceOf = renderer.create(<VerifySSN />).getInstance();
  instanceOf.setSocialSecurityNumber('123456789');
  expect(instanceOf.state.socialSecurityNumber).toEqual('123456789');
});

//  mocked function call on button click
it('Called function on button click', () => {       
  const wrapper = shallow(<VerifySSN onPress={jest.fn()} />)
  const spy = jest.spyOn(wrapper.instance(), 'onClickNext');
  wrapper.instance().forceUpdate();
  wrapper.find(GButtonComponent).at(1).simulate('press');
  expect(spy).toHaveBeenCalled();
  spy.mockClear();  
});

//  //  mocked function call on button click
//  it('should call mocked function', () => {  
//    const mockedfn = jest.fn();
//    const wrapper = shallow(<VerifySSN />);

//    expect(wrapper.find(GButtonComponent).first()).toHaveLength(1);
//    wrapper.find(GButtonComponent).first().props().onPress();
//    wrapper.find(GButtonComponent).first().simulate('press');
//    expect(mockedfn).toHaveBeenCalledTimes(1);  
  
//  });

});