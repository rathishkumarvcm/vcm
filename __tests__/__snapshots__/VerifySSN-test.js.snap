// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Verify SSN  snap shot 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#F7FAFF",
      "flex": 1,
      "width": "100%",
    }
  }
>
  <View
    accessible={true}
    clickable={false}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#FFFFFF",
        "flex": 0.12,
        "flexDirection": "row",
        "opacity": 1,
        "paddingLeft": "4%",
        "paddingRight": "4%",
        "width": "100%",
      }
    }
  >
    <View
      accessible={true}
      clickable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "height": 100.5,
          "opacity": 1,
          "width": "25%",
        }
      }
    >
      <Image
        resizeMode="contain"
        source={
          Object {
            "testUri": "../../../App/Images/logo.png",
          }
        }
        style={
          Object {
            "alignItems": "flex-start",
            "justifyContent": "flex-start",
            "width": "100%",
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "width": "12%",
        }
      }
    />
    <View
      accessible={true}
      clickable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "marginTop": 24.5,
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Object {
            "height": 40,
            "width": 56,
          }
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "black",
                "fontSize": 40,
              },
              Object {
                "backgroundColor": "transparent",
                "marginHorizontal": 5,
              },
              Object {
                "fontFamily": "EvilIcons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
          testID="iconIcon"
        >
          
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "width": "10%",
        }
      }
    />
    <View
      accessible={true}
      clickable={false}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Object {
            "height": 30,
            "width": 44,
          }
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "black",
                "fontSize": 30,
              },
              Object {
                "backgroundColor": "transparent",
                "marginHorizontal": 5,
              },
              Object {
                "fontFamily": "Ionicons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
          testID="iconIcon"
        >
          
        </Text>
      </View>
    </View>
    <View
      accessible={true}
      clickable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Object {
            "height": 30,
            "width": 44,
          }
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "black",
                "fontSize": 30,
              },
              Object {
                "backgroundColor": "transparent",
                "marginHorizontal": 5,
              },
              Object {
                "fontFamily": "Ionicons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
          testID="iconIcon"
        >
          
        </Text>
      </View>
    </View>
    <View
      accessible={true}
      clickable={false}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "flex-end",
          "justifyContent": "center",
          "opacity": 1,
        }
      }
    >
      <RNGestureHandlerButton
        collapsable={false}
        onGestureEvent={[Function]}
        onGestureHandlerEvent={[Function]}
        onGestureHandlerStateChange={[Function]}
        onHandlerStateChange={[Function]}
        rippleColor={0}
        style={
          Array [
            Object {
              "overflow": "hidden",
            },
            undefined,
          ]
        }
      >
        <View
          accessible={true}
          style={
            Object {
              "alignItems": "flex-end",
              "justifyContent": "center",
              "marginEnd": 10,
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Object {
                "height": 40,
                "width": 56,
              }
            }
          >
            <Text
              allowFontScaling={false}
              style={
                Array [
                  Object {
                    "color": "black",
                    "fontSize": 40,
                  },
                  Object {
                    "backgroundColor": "transparent",
                    "marginHorizontal": 5,
                  },
                  Object {
                    "fontFamily": "Material Icons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
              testID="iconIcon"
            >
              
            </Text>
          </View>
        </View>
      </RNGestureHandlerButton>
    </View>
  </View>
  <RCTScrollView
    style={
      Object {
        "flex": 0.85,
      }
    }
  >
    <View>
      <View
        style={
          Object {
            "marginLeft": "4%",
            "marginRight": "4%",
            "marginTop": 49.5,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "center",
              "marginBottom": 49.5,
              "marginTop": 16.5,
            }
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "#56565A",
                "flex": 0.4,
                "height": 13,
                "marginRight": "1%",
              }
            }
          />
          <View
            style={
              Object {
                "backgroundColor": "#E6E6E6",
                "flex": 0.4,
                "height": 13,
                "marginRight": "1%",
              }
            }
          />
          <View
            style={
              Object {
                "backgroundColor": "#E6E6E6",
                "flex": 0.4,
                "height": 13,
                "marginRight": "1%",
              }
            }
          />
          <View
            style={
              Object {
                "backgroundColor": "#E6E6E6",
                "flex": 0.4,
                "height": 13,
                "marginRight": "1%",
              }
            }
          />
        </View>
        <Text
          style={
            Object {
              "color": "#56565A",
              "fontSize": 52.5,
            }
          }
        >
          Verification
        </Text>
        <Text
          style={
            Object {
              "color": "#56565A",
              "fontSize": 23,
              "marginTop": 24.5,
            }
          }
        >
          Enter the details for User Verification
        </Text>
        <Text
          style={
            Object {
              "color": "#56565A",
              "fontSize": 26.5,
              "fontWeight": "bold",
              "marginTop": 49.5,
            }
          }
        >
          Social Security Number
        </Text>
        <View
          style={
            Array [
              Object {
                "backgroundColor": "#FFFFFF",
                "borderColor": "#DEDEDF",
                "borderRadius": 6.5,
                "borderWidth": 1,
                "height": 79,
                "justifyContent": "center",
                "width": "92%",
              },
              Object {
                "marginTop": 16.5,
                "width": "100%",
              },
            ]
          }
        >
          <TextInput
            allowFontScaling={true}
            autoCapitalize="none"
            autoCorrect={false}
            autoFocus={false}
            editable={true}
            errorFlag={false}
            errorText="Enter a valid Social Security Number"
            inputStyle={Object {}}
            keyboardType="numeric"
            maxLength={9}
            multiline={false}
            numberOfLines={1}
            onBlur={null}
            onChange={null}
            onChangeText={[Function]}
            onSubmitEditing={null}
            placeholder=""
            placeholderTextColor={null}
            propInputStyle={
              Object {
                "marginTop": 16.5,
                "width": "100%",
              }
            }
            rejectResponderTermination={true}
            returnKeyType="next"
            secureTextEntry={true}
            selectionColor={null}
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "color": "#B5B5B5",
                  "fontSize": 26.5,
                  "justifyContent": "center",
                  "marginLeft": 23,
                },
                Object {},
              ]
            }
            underlineColorAndroid="transparent"
            value=""
          />
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "marginBottom": 16.5,
              "marginTop": 57.5,
            }
          }
        >
          <View
            accessible={true}
            clickable={true}
            isTVSelectable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignContent": "center",
                "alignItems": "center",
                "backgroundColor": "#fff",
                "borderColor": "#61285F45",
                "borderRadius": 1,
                "borderWidth": 1,
                "height": 82,
                "justifyContent": "center",
                "marginTop": "2%",
                "opacity": 1,
                "width": "90%",
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#544A54",
                    "fontSize": 26.5,
                    "fontWeight": "bold",
                    "textAlign": "center",
                    "width": "100%",
                  },
                ]
              }
            >
              Cancel
            </Text>
          </View>
          <View
            accessible={true}
            clickable={true}
            isTVSelectable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignContent": "center",
                "alignItems": "center",
                "backgroundColor": "#544A54",
                "borderColor": "#61285F45",
                "borderRadius": 1,
                "borderWidth": 1,
                "height": 82,
                "justifyContent": "center",
                "marginTop": 33,
                "opacity": 1,
                "width": "90%",
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#fff",
                    "fontSize": 26.5,
                    "fontWeight": "bold",
                    "textAlign": "center",
                    "width": "100%",
                  },
                ]
              }
            >
              Next
            </Text>
          </View>
        </View>
      </View>
      <View>
        <View
          style={
            Object {
              "borderColor": "#DEDEDF",
              "borderTopWidth": 1,
              "marginTop": 65.5,
            }
          }
        />
        <View
          style={
            Object {
              "backgroundColor": "#FFFFFF",
            }
          }
        >
          <View
            style={
              Object {
                "marginBottom": 41,
                "marginTop": 41,
                "paddingLeft": "4%",
                "paddingRight": "4%",
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#56565A",
                  "fontSize": 29.5,
                  "fontWeight": "bold",
                }
              }
            >
              Investing involves risk including loss of principal.
            </Text>
            <Text
              style={
                Object {
                  "color": "#56565A",
                  "fontSize": 26.5,
                  "lineHeight": 33,
                  "marginTop": 24.5,
                }
              }
            >
              Victory Mutual Funds and USAA Mutual Funds are distributed by Victory Capital Advisers, Inc. (VCA). VictoryShares ETFs and VictoryShares USAA ETFs are distributed by Foreside Fund Services, LLC (Foreside). VCA and Foreside are members of FINRA and SIPC. Victory Capital Management Inc. (VCM) is the investment adviser to the Victory Mutual Funds, USAA Mutual Funds, VictoryShares ETFs and VictoryShares USAA ETFs. VCA and VCM are not affiliated with Foreside. USAA is not affiliated with Foreside, VCM, or VCA. USAA and the USAA logos are registered trademarks and the USAA Mutual Funds and USAA Investments logos are trademarks of United Services Automobile Association and are being used by Victory Capital and its affiliates under license. Victory Capital means Victory Capital Management Inc., the investment manager of the USAA 529 College Savings Plan (Plan). The Plan is distributed by Victory Capital Advisers, Inc., a broker dealer registered with FINRA and an affiliate of Victory Capital. Victory Capital and its affiliates are not affiliated with United Services Automobile Association or its affiliates. USAA and the USAA logo are registered trademarks and the USAA 529 College Savings Plan logo is a trademark of United Services Automobile Association and are being used by Victory Capital and its affiliates under license.
            </Text>
          </View>
          <View
            style={
              Object {
                "marginBottom": 41,
                "marginTop": 33,
                "paddingLeft": "4%",
                "paddingRight": "4%",
              }
            }
          >
            <Image
              resizeMode="contain"
              source={
                Object {
                  "testUri": "../../../App/Images/logo.png",
                }
              }
            />
          </View>
          <View
            style={
              Object {
                "marginLeft": "4%",
                "marginTop": 33,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#56565A",
                  "fontSize": 26.5,
                  "lineHeight": 33,
                  "marginTop": 24.5,
                }
              }
            >
              Connect with Us
            </Text>
          </View>
          <View
            style={
              Object {
                "flexDirection": "row",
                "marginBottom": 33,
                "marginLeft": "2%",
                "marginTop": 16.5,
              }
            }
          >
            <View
              style={
                Object {
                  "height": 35,
                  "width": 50,
                }
              }
            >
              <Text
                allowFontScaling={false}
                style={
                  Array [
                    Object {
                      "color": "#56565A",
                      "fontSize": 35,
                    },
                    Object {
                      "backgroundColor": "transparent",
                      "marginHorizontal": 5,
                    },
                    Object {
                      "fontFamily": "anticon",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
                testID="iconIcon"
              >
                
              </Text>
            </View>
            <View
              style={
                Object {
                  "height": 35,
                  "width": 50,
                }
              }
            >
              <Text
                allowFontScaling={false}
                style={
                  Array [
                    Object {
                      "color": "#56565A",
                      "fontSize": 35,
                    },
                    Object {
                      "backgroundColor": "transparent",
                      "marginHorizontal": 5,
                    },
                    Object {
                      "fontFamily": "anticon",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
                testID="iconIcon"
              >
                
              </Text>
            </View>
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "white",
                "flexDirection": "row",
                "height": 65.5,
                "justifyContent": "center",
                "marginLeft": "2%",
                "marginTop": 24.5,
                "width": "100%",
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#5D83AE",
                  "fontSize": 26.5,
                  "fontWeight": "bold",
                  "paddingLeft": "2%",
                  "width": "50%",
                }
              }
            >
              Privacy Policy
            </Text>
            <Text
              style={
                Object {
                  "color": "#5D83AE",
                  "fontSize": 26.5,
                  "fontWeight": "bold",
                  "paddingLeft": "2%",
                  "width": "50%",
                }
              }
            >
              Fund Documents
            </Text>
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "white",
                "flexDirection": "row",
                "height": 65.5,
                "justifyContent": "center",
                "marginLeft": "2%",
                "marginTop": 24.5,
                "width": "100%",
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#5D83AE",
                  "fontSize": 26.5,
                  "fontWeight": "bold",
                  "paddingLeft": "2%",
                  "width": "50%",
                }
              }
            >
              User Agreements
            </Text>
            <Text
              style={
                Object {
                  "color": "#5D83AE",
                  "fontSize": 26.5,
                  "fontWeight": "bold",
                  "paddingLeft": "2%",
                  "width": "50%",
                }
              }
            >
              Support
            </Text>
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#56565A",
                "height": 98.5,
                "justifyContent": "center",
                "marginTop": 33,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#FFFFFF",
                }
              }
            >
              Copyright Victory Capital Management Inc. ©2020
            </Text>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
